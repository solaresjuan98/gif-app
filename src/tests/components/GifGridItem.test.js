import React from 'react';
import { shallow } from "enzyme";
import { GifGridItem } from '../../compontents/GifGridItem';

describe('Tests on <GifGridItem />', () => {


    const title = 'a title';
    const url = 'https://localhost/algo.jpg';
    const wrapper = shallow(<GifGridItem title={title} url={url} />);
    
    test('should show the component correctly', () => {

        
        expect(wrapper).toMatchSnapshot();
    });


    test('should have a paragraph with title', () => {
        
        const paragraph = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
    })
    

    test('should have the same image equal to url and alt from props', () => {
        
        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);

    });

    test('should have animate__fadeIn', () => {
        
        const div = wrapper.find('div');

        const className = div.prop('className');
        expect(className.includes('animate__fadeIn')).toBe(true);

    });
    
    


})
