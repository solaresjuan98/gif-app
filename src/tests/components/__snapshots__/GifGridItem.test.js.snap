// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <GifGridItem /> should show the component correctly 1`] = `
<div
  className="card animate__animated animate__fadeIn "
>
  <img
    alt="a title"
    src="https://localhost/algo.jpg"
  />
  <p>
    a title
  </p>
</div>
`;
